io.gitlab.liferooter.TextPieces.yaml

```diff
@@ -0,0 +1,47 @@
+# SPDX-FileCopyrightText: 2023 Gleb Smirnov <glebsmirnov0708@gmail.com>
+#
+# SPDX-License-Identifier: CC0-1.0
+
+id: io.gitlab.liferooter.TextPieces
+
+runtime: org.gnome.Platform
+runtime-version: '46'
+
+sdk: org.gnome.Sdk
+sdk-extensions:
+- org.freedesktop.Sdk.Extension.rust-stable
+
+command: textpieces
+
+finish-args:
+- --socket=fallback-x11
+- --socket=wayland
+- --device=dri
+- --share=ipc
+- --talk-name=org.freedesktop.Flatpak
+- --filesystem=~/.local/share/textpieces
```

Comment:
Shouldn't this be `xdg-data/textpieces`?

Reply:
No, it shouldn't. In some old days when the hack was created I used `~/.local/share/textpieces` not to use `~/.var/app/<id>/data/textpieces` to make it possible to edit scripts with other apps when it was impossible to edit one app's data with another one. It was probably a mistake, it's was better just to use `--filesystem=xdg-data/textpieces` to do the same thing, but without hardcoding the path. Now this permission is needed to move data away from this hacky place to where it should be. In some future this permission will be removed and users who has not migrated for too long will have to migrate manually. But not now.

Reply:
xdg-data resolves to `~/.local/share` by default (if the user has not set `XDG_USER_DATA`), not `~/.var/app/<id>/data`.

Reply:
Yes, I know. And that's why it would be better to use it in the past where I moved the script directory out of the app data directory. But now I'm using this permission to migrate the data from that directory. So I hardcode the directory because it was hardcoded in the previous versions. I do not the app to follow `XDG_DATA_HOME` variable because if it follows, it finds no data because the old one stored it in the hardcoded directory.